boards/arm/thingy52_nrf52832/board.c

```diff
@@ -24,7 +24,28 @@ static int pwr_ctrl_init(void)
 		return ret;
 	}
 
-	k_sleep(K_MSEC(1)); /* Wait for the rail to come up and stabilize */
+	k_sleep(K_MSEC(100)); /* Wait for the rail to come up and stabilize */
+
+	return 0;
+}
+
+static const struct gpio_dt_spec mpu_gpio =
+	GPIO_DT_SPEC_GET(DT_NODELABEL(mpu_pwr), enable_gpios);
+
+static int mpu_pwr_ctrl_init(void)
```

Comment:
Instead you can update the `status` property of the `mpu_pwr` node in the devicetree